plugins/SpectrumAnalyzer/SaProcessor.h

```diff
@@ -42,19 +45,35 @@ class SaProcessor
 	explicit SaProcessor(SaControls *controls);
 	virtual ~SaProcessor();
 
-	void analyse(sampleFrame *in_buffer, const fpp_t frame_count);
+	// analysis thread and a method to terminate it
+	void analyze(ringbuffer_t<sampleFrame> &ring_buffer, QWaitCondition &notifier);
+	void terminate() {m_terminate = true;}
 
 	// inform processor if any processing is actually required
 	void setSpectrumActive(bool active);
 	void setWaterfallActive(bool active);
+	void flipRequest() {m_flipRequest = true;}	// request refresh of history buffer
 
 	// configuration is taken from models in SaControls; some changes require
 	// an exlicit update request (reallocation and window rebuild)
 	void reallocateBuffers();
 	void rebuildWindow();
 	void clear();
```

Comment:
The latter two could definitely be useful for display classes. I'm not sure about "clear", I don't have a use-case for it now, but it could be useful to have some way of getting the processing unit to a known "initial state" from the user point of view, so it probably makes sense keeping it public as well.